imblearn/utils/validation.py

```diff
@@ -33,9 +45,281 @@ def check_neighbors_object(nn_name, nn_object, additional_neighbor=0):
     nn_object : KNeighborsMixin
         The k-NN object.
     """
-    if isinstance(nn_object, int):
+    if isinstance(nn_object, Integral):
         return NearestNeighbors(n_neighbors=nn_object + additional_neighbor)
     elif isinstance(nn_object, KNeighborsMixin):
         return nn_object
     else:
         raise_isinstance_error(nn_name, [int, KNeighborsMixin], nn_object)
+
+
+def check_target_type(y):
+    """Check the target types to be conform to the current samplers.
+
+    The current samplers should be compatible with ``'binary'`` and
+    ``'multiclass'`` targets only.
+
+    Parameters
+    ----------
+    y : ndarray,
+        The array containing the target
+
+    Returns
+    -------
+    y : ndarray,
+        The returned target.
+
+    """
+    if type_of_target(y) not in TARGET_KIND:
+        # FIXME: perfectly we should raise an error but the sklearn API does
+        # not allow for it
+        warnings.warn("'y' should be of types {} only. Got {} instead.".format(
+            TARGET_KIND, type_of_target(y)))
+    return y
+
+
+def hash_X_y(X, y):
+    """Compute hash of the input arrays.
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_samples, n_features)
+        The ``X`` array.
+
+    y : ndarray, shape (n_samples)
+
+    Returns
+    -------
+    X_hash: str
+        Hash identifier of the ``X`` matrix.
+
+    y_hash: str
+        Hash identifier of the ``y`` matrix.
+    """
+    return joblib.hash(X), joblib.hash(y)
```

Comment:
it needs to be profiled. But I think that this is too slow. We could randomly pick some samples which should be better than the shape. But I am still unsure about that part.